pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -350,7 +356,7 @@ func (r *controlPlaneInstanceReconciler) Reconcile(ctx context.Context) (result
 	r.Status.ChartVersion = r.chartVersion
 	updateControlPlaneConditions(r.Status, nil)
 
-	reconciliationComplete = true
+	hacks.SkipReconciliationUntilCacheSynced(ctx, common.ToNamespacedName(r.Instance))
```

Comment:
won't this move out reconcilation always by 2 seconds?

Reply:
Yes, but not always. Only when the SMCP status is updated, in which case the operator shouldn't reconcile the same SMCP again until all the watch events are received. Since there's no 100% solution for this, we use the 2s delay.